Lib/glyphsLib/interpolation.py

```diff
@@ -86,33 +86,39 @@ def build_designspace(masters, master_dir, out_dir, instance_data):
 def add_masters_to_writer(writer, ufos):
     """Add master UFOs to a MutatorMath document writer.
 
-    Returns the masters' base family and style names, as determined by taking
-    the intersection of their individual family/style names. This is used for
+    Returns the masters' family name and shared style names. These are used for
     naming instances and the designspace path.
     """
 
-    base_family = ''
-    base_style = ''
-    specify_info_source = True
+    master_data = []
+    base_family = None
+    base_style = None
 
     for font in ufos:
         family, style = font.info.familyName, font.info.styleName
-        if family in base_family or not base_family:
+        if base_family is None:
             base_family = family
-        elif base_family not in family:
-            raise ValueError('Inconsistent family names for masters')
-        if style in base_style or not base_style:
-            base_style = style
+        assert family == base_family, 'Masters must all have same family name'
+        if base_style is None:
+            base_style = style.split()
+        base_style = [s for s in style.split() if s in base_style]
+        master_data.append((font.path, family, style, {
+            s: font.lib.get(GLYPHS_PREFIX + s + 'Value', DEFAULT_LOC)
+            for s in ('weight', 'width', 'custom')}))
+
+    base_style = ' '.join(base_style)
+    info_source = 0
+    for i, (path, family, style, location) in enumerate(master_data):
+        if family == base_family and style == (base_style or 'Regular'):
+            info_source = i
+            break
+
+    for i, (path, family, style, location) in enumerate(master_data):
```

Comment:
Not sure why you loop over master_data twice.  is_base is false until the family and style match, then it is true that one time, then it is false thereafter.